src/lib/simulation/simulation.ts

```diff
@@ -20,6 +20,9 @@ export const simulateEVMTransactionWithAlchemy = async (
   quantity: number,
   blockNumber?: string,
 ): Promise<{ message: string; success: boolean; rawSimulationResult: any }> => {
```

Comment:
This is all code that is duplicated from Ecommerce. Ecomm has even more logic around this in `constructWeb3Payload`